eth/protocols/eth/handlers.go

```diff
@@ -228,7 +229,33 @@ func ServiceGetBlockBodiesQuery(chain *core.BlockChain, query GetBlockBodiesRequ
 		if body == nil {
 			continue
 		}
+		header := chain.GetHeaderByHash(hash)
+		if header == nil {
+			continue
+		}
+
 		sidecars := chain.GetSidecarsByHash(hash)
+		// Only check for sidecars validity when necessary
```

Comment:
I think we don't need these logic,
expired sidecars will automaticly be pruned,
so chain.GetSidecarsByHash(hash) will get sidecars if needed, otherwise get nil (also is expected).

Reply:
Yeah but the issue then, for when sidecars are actually needed, is that the peer will be considered invalid if it doesn't send the block w/ sidecars. While if it just returns `nil` from the request, it won't be marked as invalid, and the block will be fetched from another peer.